src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -0,0 +1,181 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+import os
+from six.moves import configparser
+
+from azure.cli.core._config import GLOBAL_CONFIG_DIR
+
+CUSTOM_CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
+
+class CloudNotRegisteredException(Exception):
+    def __init__(self, cloud_name):
+        super(CloudNotRegisteredException, self).__init__(cloud_name)
+        self.cloud_name = cloud_name
+    def __str__(self):
+        return "The cloud '{}' is not registered.".format(self.cloud_name)
+
+class CloudAlreadyRegisteredException(Exception):
+    def __init__(self, cloud_name):
+        super(CloudAlreadyRegisteredException, self).__init__(cloud_name)
+        self.cloud_name = cloud_name
+    def __str__(self):
+        return "The cloud '{}' is already registered.".format(self.cloud_name)
+
+class CannotUnregisterCloudException(Exception):
+    pass
+
+class CloudEndpointUrl(object): # pylint: disable=too-few-public-methods
+    MANAGEMENT = 'management'
```

Comment:
The class isn't used in this way.
It's more of a container for storing the key names for the cloud endpoints. Kind of like an enum with the value being a string instead of an int.

I don't call the constructor anywhere.